packages/zoe/src/zoeService/startInstance.js

```diff
@@ -191,10 +191,14 @@ export const makeStartInstance = (
 
         return E(state.adminNode).done();
       },
-      restartContract(_newPrivateArgs = undefined) {
+      restartContract(newPrivateArgs = undefined) {
         const { state } = this;
 
-        const vatParameters = { contractBundleCap: state.contractBundleCap };
+        const vatParameters = {
+          contractBundleCap: state.contractBundleCap,
+          privateArgs: newPrivateArgs,
```

Comment:
I wonder about a test. I suppose this is tested somewhere else, if indirectly...

Reply:
yeah the upgrade test failed with it. agree a unit test would be better but I propose to led it slide since it's unlikely to break without CI balking